az-cvm-vtpm/az-tdx-vtpm/src/lib.rs

```diff
@@ -1,38 +1,52 @@
 // Copyright (c) Microsoft Corporation.
 // Licensed under the MIT License.
 
-//!  This library enables guest attestation flows for [TDX CVMs on Azure](https://learn.microsoft.com/en-us/azure/confidential-computing/tdx-confidential-vm-overview). TDX CVMs are currently in limited preview and hence the library is considered experimental and subject to change.
+//!  This library enables guest attestation flows for [TDX CVMs on Azure](https://learn.microsoft.com/en-us/azure/confidential-computing/tdx-confidential-vm-overview).
+//!  The library provides a function to retrieve the TDX quote from the Azure Instance Metadata Service (IMDS).
+//!  The code below illustrates how to use the library to retrieve the TDX quote.
+//!  
 //!  #
 //!  ```no_run
-//!  use az_tdx_vtpm::{imds, hcl, vtpm};
+//!  use az_tdx_vtpm::{imds, report};
 //!  use std::error::Error;
 //!
 //!  fn main() -> Result<(), Box<dyn Error>> {
-//!    let bytes = vtpm::get_report()?;
-//!    let hcl_report = hcl::HclReport::new(bytes)?;
-//!    let tdx_report_slice = hcl_report.tdx_report_slice();
-//!    let report_body = imds::ReportBody::new(tdx_report_slice);
-//!    let td_quote_bytes = imds::get_td_quote(report_body)?;
-//!    let hash = hcl_report.var_data_sha256();
-//!    println!("var_data hash: {:x?}", hash);
```

Comment:
Why is this removed from the example? Also, shouldn't we check the hash and compare it with the td_report here similar to what we do in SNP?

Reply:
mmh, right, it would be good to add verification sample code to a common example project, but so far the example code only covers SNP so it makes sense to add it to the docs.